src/table.js

```diff
@@ -538,15 +543,99 @@ export function getTypeValidator(colType) {
   }
 }
 
+export function coerceToType(value, type, options = {}) {
+  const defaultValue = options.soft ? value : null;
+  const numberDefault = defaultValue === null ? NaN : defaultValue;
+  switch (type) {
+    case "string": 
+      return typeof value === "string"
+        ? value.trim()
+        : value || value === 0
+        ? value.toString()
+        : defaultValue;
+    case "boolean":
+      return value === true || value === "true"
+        ? true
+        : value === false || value === "false"
+        ? false
+        : defaultValue;
+    case "integer":
+      return !value || isNaN(parseInt(value)) ? numberDefault : parseInt(value);
+    case "bigint":
+      return typeof value === "bigint"
+        ? value
+        : !value || isNaN(value) || !Number.isInteger(+value)
```

Comment:
This `!value` check creates unexpected behavior for the number 0 (which I believe should be coercible to the BigInt `0n` via `BigInt(0)`. Is the `isNaN` check sufficient, or is `!value` catching certain values like `null` or `undefined`?